provider/cmd/pulumi-resource-nuage/nuage_provider/container_function.py

```diff
@@ -114,21 +116,52 @@ def __init__(self, name: str, args: ContainerFunctionArgs, props: Optional[dict]
         
         if args.repository:
             # Get existing repository using repository name.
-            repository = aws.ecr.get_repository(name=args.repository).repository_url
+            repository_url = aws.ecr.get_repository(name=args.repository).repository_url
         else:
             # If repository is not defined, create new ecr repository using `ecr_repository_name`.
             repository = awsx.ecr.Repository(
                 resource_name=f"{args.ecr_repository_name}-repository",
                 name=args.ecr_repository_name,
-            ).url
+            )
+            repository_url = repository.url
+
+            repository_lifecycle = aws.ecr.LifecyclePolicy(
+                f"{args.ecr_repository_name}-ecr-lifecycle",
```

Comment:
Please use `resource_name` as a prefix for every resource in the ContainerFunction : this makes it easier to track deployed resources in the AWS Console